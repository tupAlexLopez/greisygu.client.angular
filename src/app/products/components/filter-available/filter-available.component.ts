import { Component, EventEmitter, Output } from '@angular/core';

@Component({
  selector: 'product-filter-available',
  templateUrl: './filter-available.component.html',
  styleUrls: ['./filter-available.component.css']
})
export class FilterAvailableComponent {
  @Output() eventAvailable = new EventEmitter<boolean>();

  onClickItem( value:boolean ):void {
    this.eventAvailable.emit( value );
  }
}
