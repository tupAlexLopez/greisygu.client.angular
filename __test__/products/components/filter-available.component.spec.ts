import { ComponentFixture, TestBed } from '@angular/core/testing';
import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
import { HarnessLoader } from '@angular/cdk/testing';
import { MatFormFieldHarness } from '@angular/material/form-field/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ReactiveFormsModule } from '@angular/forms';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatSelectModule } from '@angular/material/select';

import { FilterAvailableComponent } from '../../../src/app/products/components/filter-available/filter-available.component';

describe('FilterAvailableComponent', () => {
  let component: FilterAvailableComponent;
  let fixture: ComponentFixture<FilterAvailableComponent>;
  let compiled: HTMLElement;
  let loader: HarnessLoader;

  beforeEach(async() => {
    await TestBed.configureTestingModule({
      declarations: [ FilterAvailableComponent ],
      imports: [
        ReactiveFormsModule,
        MatFormFieldModule,
        MatSelectModule,
        NoopAnimationsModule
      ]
    })
    .compileComponents();
  });

  beforeEach(()=> {
    fixture = TestBed.createComponent(FilterAvailableComponent);
    component = fixture.componentInstance;
    loader = TestbedHarnessEnvironment.loader( fixture );
    compiled = fixture.nativeElement;
    
    fixture.detectChanges();
  });

  test('Deberia inicializar el componente', () => {
    expect(component).toBeTruthy();
  });

  test('Debe coincidir el HTML con el snapshot', ()=> {
    const html = compiled.innerHTML;
    expect( html ).toMatchSnapshot();
  });

  test('HTML debe tener un mat-form-field con su respectivo label.', async()=> {
    const labelTextExpected:string = 'Disponibilidad';
    const form = await loader.getHarness<MatFormFieldHarness>(MatFormFieldHarness);
    const label = await form.getLabel();

    expect( form ).not.toBeNull();
    expect( label ).toEqual( labelTextExpected );
  })

  test('HTML debe tener un mat-select con sus respectivas opciones disponibles.', async()=> {
    
  })

  

});
