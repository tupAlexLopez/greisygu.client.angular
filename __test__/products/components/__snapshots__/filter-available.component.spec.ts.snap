// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterAvailableComponent Debe coincidir el HTML con el snapshot 1`] = `
"<mat-form-field class="mat-mdc-form-field ng-tns-c4-0 mat-mdc-form-field-type-mat-select mat-form-field-no-animations mat-form-field-appearance-fill mat-form-field-hide-placeholder mat-primary"><!--container--><div class="mat-mdc-text-field-wrapper mdc-text-field ng-tns-c4-0 mdc-text-field--filled"><div class="mat-mdc-form-field-focus-overlay ng-tns-c4-0 ng-star-inserted"></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--><div class="mat-mdc-form-field-flex ng-tns-c4-0"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div class="mat-mdc-form-field-infix ng-tns-c4-0"><label matformfieldfloatinglabel="" class="mdc-floating-label mat-mdc-floating-label ng-tns-c4-0 ng-star-inserted" ng-reflect-floating="false" ng-reflect-disabled="true" id="mat-mdc-form-field-label-2" for="mat-select-2" aria-owns="mat-select-2"><mat-label class="ng-tns-c4-0">Disponibilidad</mat-label><!--bindings={
  "ng-reflect-ng-if": "false"
}--></label><!--bindings={
  "ng-reflect-ng-if": "true"
}--><!--bindings={
  "ng-reflect-ng-template-outlet": "[object Object]"
}--><!--bindings={
  "ng-reflect-ng-if": "true"
}--><mat-select role="combobox" aria-autocomplete="none" aria-haspopup="listbox" class="mat-mdc-select ng-tns-c6-1 ng-tns-c4-0 mat-mdc-select-empty ng-star-inserted" aria-labelledby="mat-mdc-form-field-label-2 mat-select-value-3" id="mat-select-2" tabindex="0" aria-expanded="false" aria-required="false" aria-disabled="false" aria-invalid="false"><div cdk-overlay-origin="" class="mat-mdc-select-trigger ng-tns-c6-1"><div class="mat-mdc-select-value ng-tns-c6-1" ng-reflect-ng-switch="true" id="mat-select-value-3"><span class="mat-mdc-select-placeholder mat-mdc-select-min-line ng-tns-c6-1 ng-star-inserted"></span><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}--></div><div class="mat-mdc-select-arrow-wrapper ng-tns-c6-1"><div class="mat-mdc-select-arrow ng-tns-c6-1"><svg viewBox="0 0 24 24" width="24px" height="24px" focusable="false" class="ng-tns-c6-1"><path d="M7 10l5 5 5-5z" class="ng-tns-c6-1"></path></svg></div></div></div><!--bindings={}--></mat-select></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--></div><div matformfieldlineripple="" class="mdc-line-ripple ng-tns-c4-0 mdc-line-ripple--deactivating ng-star-inserted"></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></div><div class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align ng-tns-c4-0" ng-reflect-ng-switch="hint"><!--bindings={
  "ng-reflect-ng-switch-case": "error"
}--><div class="mat-mdc-form-field-hint-wrapper ng-tns-c4-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted" style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"><!--bindings={
  "ng-reflect-ng-if": ""
}--><div class="mat-mdc-form-field-hint-spacer ng-tns-c4-0"></div></div><!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}--></div></mat-form-field>"
`;
